js/ui/map.js

```diff
@@ -202,6 +203,11 @@ var Map = module.exports = function(options) {
     this.resize();
 
     if (options.classes) this.setClasses(options.classes);
+    if (options.customSourceTypes) {
+        for (var sourceTypeName in options.customSourceTypes) {
+            Source.addType(sourceTypeName, options.customSourceTypes[sourceTypeName]);
+        }
+    }
```

Comment:
Sources can be added either by `mapboxgl.addSourceType` or the `customSourceTypes` constructor option. Can we settle on a single API?